socketio/src/asynchronous/client/client.rs

```diff
@@ -162,7 +162,21 @@ impl Client {
                 drop(stream);
 
                 let should_reconnect = match *(client_clone.disconnect_reason.read().await) {
-                    DisconnectReason::Unknown => reconnect,
+                    DisconnectReason::Unknown => {
+                        // If we disconnected for an unknown reason, the client might not have noticed
+                        // the closure yet. Hence, fire a transport close event to notify it.
+                        // We don't need to do that in the other cases, since proper server close
+                        // and manual client close are handled explicitly.
+                        if let Some(err) = client_clone
+                            .callback(&Event::Close, "transport close")
```

Comment:
nit: constant for this message

Reply:
I added a new `CloseReason` enum and used the appropriate values wherever the `Event::Close` was emitted. This should make it more consistent and easier for clients to distinguish the close reasons.